app/javascript/article-form/components/EditorBody.jsx

```diff
@@ -1,14 +1,61 @@
 import { h } from 'preact';
 import PropTypes from 'prop-types';
 import Textarea from 'preact-textarea-autosize';
+import { useEffect, useRef } from 'preact/hooks';
 import { Toolbar } from './Toolbar';
+import {
+  handleImageDrop,
+  handleImageFailure,
+  onDragOver,
+  onDragExit,
+} from './dragAndDropHelpers';
+import { useDragAndDrop } from '@utilities/dragAndDrop';
+
+function handleImageSuccess(textAreaRef) {
+  return function (response) {
+    // Function is within the component to be able to access
+    // textarea ref.
+    const editableBodyElement = textAreaRef.current.base;
+    const { links, image } = response;
+
+    const markdownImageLink = `![${image[0].name}](${links[0]})`;
+    const { selectionStart, selectionEnd, value } = editableBodyElement;
+    const before = value.substring(0, selectionStart);
+    const after = value.substring(selectionEnd, value.length);
+
+    editableBodyElement.value = `${before + markdownImageLink} ${after}`;
```

Comment:
as we don't really support inline images, should the insertion have a "newline" as well?

I tried to upload two images and the tags were kept on the same line

Reply:
I'm fine with adding a new line. Coming up!